### PR TITLE
Fix device in atomistic model check

### PR DESCRIPTION
**Pull request by frostedoyster**
This fixes a small device bug in the atomistic model checks. The predictions can be on any device, but they were being compared to a CPU Labels object.
I am not sure how to test this however

# Contributor (creator of pull-request) checklist

 - [?] Tests updated (for new features and bugfixes)?
 - [ ] Documentation updated (for new features)?
 - [ ] Issue referenced (for PRs that solve an issue)?

# Reviewer checklist

 - [ ] CHANGELOG updated with public API or any other important changes?

----
📚 Documentation preview 📚: https://metatensor--552.org.readthedocs.build/en/552/

### CODE REVIEW

File: python/metatensor-torch/metatensor/torch/atomistic/outputs.py [hunk 1]
```diff
@@ -75,7 +75,9 @@ def _check_energy(
         if selected_atoms is not None:
             expected_samples = expected_samples.intersection(selected_atoms)
 
-        if len(expected_samples.union(energy_block.samples)) != len(expected_samples):
+        if len(expected_samples.union(energy_block.samples.to(device="cpu"))) != len(
```
Comment (Luthaf) [inline python/metatensor-torch/metatensor/torch/atomistic/outputs.py:78]: instead of moving the data to CPU, I would rather create the `expected_samples` on the right device, replacing `torch.tensor(expected_values)` with `torch.tensor(expected_values, device=energy.device)`

File: python/metatensor-torch/metatensor/torch/atomistic/outputs.py [hunk 2]
```diff
@@ -71,7 +71,10 @@ def _check_energy(
             for a in range(len(system)):
                 expected_values.append([s, a])
 
-        expected_samples = Labels(["system", "atom"], torch.tensor(expected_values))
+        expected_samples = Labels(
+            ["system", "atom"],
+            torch.tensor(expected_values, device=energy_block.samples.values.device),
```
Comment (Luthaf) [inline python/metatensor-torch/metatensor/torch/atomistic/outputs.py:76]:
```suggestion
            torch.tensor(expected_values, device=energy_block.values.device),
```

**Review by Luthaf — APPROVED** [review]
